### PR TITLE
add CURL recipe from `LibCURL`

### LINKED ISSUE #2720: Builder Request: a `Curl_jll.jl` package that includes the `curl` binary executable
From https://github.com/JuliaLang/julia/issues/40041#issuecomment-799577812:
> Perhaps we should create a separate CURL_jll that provides the executable, for users that want to be able to use it.

Since the Lib_CURL_jll stdlib in Julia 1.6 won't have the `curl` executable, it would be great to have a `CURL_jll` package for users that need the `curl` executable.

cc: @staticfloat @giordano

### PR DESCRIPTION
**Pull request by ericphanson**
closes #2720

### CODE REVIEW

**Comment by giordano** [comment]
Wait, this is _identical_ to `LibCURL_jl` and that package already ships the curl binary?  That's bad.  We don't want duplicates like that, this should _depend_ on the same version of `LibCURL_jll` and provide only the binaries.  Something similar to [`HarfBuzz`](https://github.com/JuliaPackaging/Yggdrasil/blob/3b556643f0220f8e918938b5a203cba96beb4dc1/H/HarfBuzz/common.jl).

**Comment by DilumAluthge** [comment]
@staticfloat @giordano What's the best way to move this forward?

**Comment by DilumAluthge** [comment]
Bump @staticfloat @giordano

**Comment by giordano** [comment]
I think I [explained already](https://github.com/JuliaPackaging/Yggdrasil/pull/4153#issuecomment-1004770913) what should be done?

File: C/CURL/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # On Darwin, we need to use SecureTransport (native TLS library)
+    FLAGS+=(--with-secure-transport)
+
+    # We need to explicitly request a higher `-mmacosx-version-min` here, so that it doesn't
+    # complain about: `Symbol not found: ___isOSVersionAtLeast`
+    if [[ "${target}" == aarch64* ]]; then
+        export CFLAGS=-mmacosx-version-min=11.0
+    else
+        export CFLAGS=-mmacosx-version-min=10.11
+    fi
```
Comment (giordano) [inline C/CURL/build_tarballs.jl:49]: The aarch64 branch for sure isn't needed anymore.  Not sure about the x86_64 one, might still be needed (by default it'd be `-mmacosx-version-min=10.10`)

**Comment by DilumAluthge** [comment]
Alright, here's my attempt: https://github.com/JuliaPackaging/Yggdrasil/pull/4472

I tried to follow the HarfBuzz example.

**Pull request closed without merging**